WalletWasabi.Tests/UnitTests/EventSourcing/InMemoryEventRepositoryTests.cs

```diff
@@ -0,0 +1,452 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.EventSourcing;
+using WalletWasabi.Exceptions;
+using WalletWasabi.Extensions;
+using WalletWasabi.Interfaces.EventSourcing;
+using WalletWasabi.Tests.UnitTests.EventSourcing.TestDomain;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace WalletWasabi.Tests.UnitTests.EventSourcing
+{
+	public class InMemoryEventRepositoryTests
+	{
+		private const string TestRoundAggregate = "TestRoundAggregate";
+		private readonly TimeSpan _semaphoreWaitTimeout = TimeSpan.FromSeconds(1);
+
+		public InMemoryEventRepositoryTests(ITestOutputHelper output)
+		{
+			Output = output;
+			TestEventRepository = new TestInMemoryEventRepository(output);
+			EventRepository = TestEventRepository;
+		}
+
+		private IEventRepository EventRepository { get; init; }
+		private TestInMemoryEventRepository TestEventRepository { get; init; }
+
+		private ITestOutputHelper Output { get; init; }
+
+		[Fact]
+		public async Task AppendEvents_Zero_Async()
+		{
+			// Arrange
+			var events = Array.Empty<WrappedEvent>();
+
+			// Act
+			await EventRepository.AppendEventsAsync(nameof(TestRoundAggregate), "1", events);
+
+			// Assert
+			Assert.True((await EventRepository.ListEventsAsync(nameof(TestRoundAggregate), "1"))
+				.SequenceEqual(events));
```

Comment:
what does that mean? Is it passive aggressive that I have to change or it won't be merged or can it stay?

Reply:
No, you don't have to do it. I am just expressing preference and how the rest of the code looks like.

Reply:
👍

Reply:
👍 on that suggestion. 

```cs
Assert.True((await EventRepository.ListEventsAsync(nameof(TestRoundAggregate), "1"))
    .SequenceEqual(events));
```

is not that great because one has to focus quite a bit to verify the parentheses are correctly put otherwise one can end up with something like [this](https://github.com/zkSNACKs/WalletWasabi/pull/6724#discussion_r760863535).

PS: I would say that it's the other way - i.e. using your own code style (compare your code and a few files from this project) instead of using the existing one is PA imho.

Reply:
separate lines for what? I have in on two lines 🤷‍♂️ Can you make to code suggestion here? 🙏

Reply:
and you said luca's word is final. Lucas said: "No, you don't have to do it. I am just expressing preference and how the rest of the code looks like." So again I'm confused. Lucas likes it.